File: 69_sqr_x.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left , right = 0 , x
        result = -1
        while(left <= right):
            middle = left + (right - left) // 2
            sqr = middle * middle
            if(sqr < x):
                result = middle
                left = middle + 1
            elif(sqr > x):
                right = middle -1
            else:
                result = middle
                return result
        return result

File: 69_sqr_x.c
int mSqrt(int x)
{
    int left = 0, right = x;
    int result = -1;
    while (left <= right)
    {
        long int middle = left + (right - left) / 2;
        if (middle * middle <= x)
        {
            result = middle;
            left = middle + 1;
        }
        else
        {
            right = middle - 1;
        }
    
    }
    return result;
}